tpmutil/structures.go

```diff
@@ -14,13 +14,84 @@
 
 package tpmutil
 
-import "io"
+import (
+	"encoding/binary"
+	"io"
+)
 
 // RawBytes is for Pack and RunCommand arguments that are already encoded.
 // Compared to []byte, RawBytes will not be prepended with slice length during
 // encoding.
 type RawBytes []byte
 
+// U16Bytes is a byte slice with a 16-bit header
+type U16Bytes []byte
+
+// TPMMarshal packs U16Bytes
+func (b *U16Bytes) TPMMarshal(out io.Writer) error {
+	size := uint16(len([]byte(*b)))
+	if err := binary.Write(out, binary.BigEndian, size); err != nil {
+		return err
+	}
+	if err := binary.Write(out, binary.BigEndian, []byte(*b)); err != nil {
```

Comment:
`return binary.Write(out, binary.BigEndian, []byte(*b))`
similar for all methods below

Also, is `[]byte` conversion here necessary?